src/operator/batch_norm.cc

```diff
@@ -98,7 +98,7 @@ void BatchNormOp<xpu, DType, AccReal>::DoForward(mshadow::Stream<cpu> *,
   const size_t itemCountPerChannel = inputData.Size() / channelCount;
 
   #pragma omp parallel for
-  for (int channel = 0; channel < channelCount; ++channel) {
+  for (index_t channel = 0; channel < channelCount; ++channel) {
```

Comment:
Fixed this by casting the loop condition. Created new variables because casting on each loop iteration overhead can be avoided this way.